server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzer.java

```diff
@@ -0,0 +1,744 @@
+        final int maxDocs = reader.maxDoc();
+        for (FieldInfo field : reader.getFieldInfos()) {
+            final DocValuesType dvType = field.getDocValuesType();
+            if (dvType == DocValuesType.NONE) {
+                continue;
+            }
+            cancellationChecker.checkForCancellation();
+            directory.resetBytesRead();
+            switch (dvType) {
+                case NUMERIC:
+                    iterateDocValues(maxDocs, () -> docValuesReader.getNumeric(field), NumericDocValues::longValue);
+                    break;
+                case SORTED_NUMERIC:
+                    iterateDocValues(maxDocs, () -> docValuesReader.getSortedNumeric(field), dv -> {
+                        for (int i = 0; i < dv.docValueCount(); i++) {
+                            cancellationChecker.logEvent();
+                            dv.nextValue();
+                        }
+                    });
+                    break;
+                case BINARY:
+                    iterateDocValues(maxDocs, () -> docValuesReader.getBinary(field), BinaryDocValues::binaryValue);
+                    break;
+                case SORTED:
+                    SortedDocValues sorted = iterateDocValues(maxDocs, () -> docValuesReader.getSorted(field), SortedDocValues::ordValue);
+                    sorted.lookupOrd(0);
+                    sorted.lookupOrd(sorted.getValueCount() - 1);
+                    break;
+                case SORTED_SET:
+                    SortedSetDocValues sortedSet = iterateDocValues(maxDocs, () -> docValuesReader.getSortedSet(field), dv -> {
+                        while (dv.nextOrd() != SortedSetDocValues.NO_MORE_ORDS) {
+                            cancellationChecker.logEvent();
+                        }
+                    });
+                    sortedSet.lookupOrd(0);
+                    sortedSet.lookupOrd(sortedSet.getValueCount() - 1);
+                    break;
+                default:
+                    assert false : "Unknown docValues type [" + dvType + "]";
+                    throw new IllegalStateException("Unknown docValues type [" + dvType + "]");
+            }
+            stats.addDocValues(field.name, directory.getBytesRead());
+        }
+    }
+
+    private void readProximity(Terms terms, PostingsEnum postings) throws IOException {
+        postings.freq();
+        if (terms.hasPositions()) {
+            for (int pos = 0; pos < postings.freq(); pos++) {
+                postings.nextPosition();
+                postings.startOffset();
+                postings.endOffset();
+                postings.getPayload();
+            }
+        }
+    }
+
+    private BlockTermState getBlockTermState(TermsEnum termsEnum, BytesRef term) throws IOException {
+        if (term != null && termsEnum.seekExact(term)) {
+            final TermState termState = termsEnum.termState();
+            if (termState instanceof Lucene84PostingsFormat.IntBlockTermState) {
```

Comment:
also support `Lucene50PostingsFormat.IntBlockTermState`?

Reply:
added in https://github.com/elastic/elasticsearch/pull/74051/commits/3a38efa55ecda265250768f585bad3abe7e2b1d1